flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DispatcherResourceManagerComponent.java

```diff
@@ -88,6 +95,18 @@
 
 	private void registerShutDownFuture() {
 		FutureUtils.forward(dispatcherRunner.getShutDownFuture(), shutDownFuture);
+		BiConsumer<Object, Throwable> terminateAction = (ignored, throwable) -> {
+			if (throwable != null) {
+				shutDownFuture.completeExceptionally(throwable);
+			} else {
+				shutDownFuture.complete(ApplicationStatus.SUCCEEDED);
+			}
+			if (isRunning.get()) {
+				fatalErrorHandler.onFatalError(throwable);
+			}
+		};
+		dispatcherRunner.getTerminationFuture().whenComplete(terminateAction);
```

Comment:
I think the `DispatcherRunner.getTerminationFuture` won't complete if the `Dispatcher` terminates. The problem is that we only forward the result of the `DispatcherLeaderProcess.terminationFuture` to the `DispatcherRunner.terminationFuture` if we call `closeAsync`. I would suggest to add tests to verify the intended behavior.

Reply:
But how we get the Dispatcher's TerminationFuture in DispatcherResourceManagerComponent?
Please give a hint, thanks

Reply:
@tillrohrmann  I need some help,  I took a look at this piece of code,  `DispatcherRunner.terminationFuture` is not connected to `Dispatcher`, so how we get the `Dispatcher.TerminationFuture` in `DispatcherResourceManagerComponent`?
Please give a hint, thanks

Reply:
@tillrohrmann is on vacation for a few weeks starting from today. He won't be able to resolve your questions.

I would suggest to follow his suggestions and add test cases reproducing the problem first. This should make it easier finding the right location in the code in the code handle errors.

Reply:
@rmetzger  thanks, I try to solve it myself